src/game.cpp

```diff
@@ -10460,6 +10460,24 @@ bool game::plmove( int dx, int dy, int dz )
         return true;
     }
 
+    //Wooden Fence Gate: vault over it if we are running, open it if we are walking
+    if( m.passable_ter_furn( dest_loc ) && m.open_door( dest_loc, !m.is_outside( u.pos() ), true ) ) {
```

Comment:
I wanted to have a message there specifically because the wooden fence easily overlooked and this action prevents the player from moving in the same turn (to give the player the reason why he didn't traverse a tile). It's different for normal doors but these block vision so they are much more apparent.

I would not oppose to simplifying further, but I think there is gain in explicity. In the current case it is easy to see how each condition shapes code flow. Honestly I am surprised that you guys opt for state changing methods within if conditions (such as m.open_door). While that sure reduces the number of lines of code and arguably simplifies it, it obstructs code flow visibility. But this might only be my impression that I got from reading into the codebase.

If you want me to simplify it to be most implicit, I can surely do it, but future contributors should in my opinion clearly see the intention of this piece of code. I can however add some more commentary while applying your change so the idea stays visible.

Reply:
I do prefer the simplified approach but it's not a strong opinion.  I can see your argument also.  With luck you might get someone else to offer an opinion, although that probably won't happen now this string of comments is marked outdated.

With regard to actions-within-ifs, I conjecture (as a relatively recent arrival at the codebase) this is primarily about reducing code duplication.  The alternative is to test whether something is possible before doing it, but that tends to lead to all the conditions being checked twice (once in the test-possibility codepath and again as a sanity check or because you need the same supporting data in the actually-doing-it codepath).  This can be somewhat ameliorated by techniques such as the `open_door` function which serves both purposes depending on the `bool` flag, but it's still not ideal.

Reply:
I left the passable-terrain check in, to make sure this doesn't get in the way of anything else. See commit 1e97b29